### DISCUSSION TITLE
v1.0.0 🎉

### DISCUSSION THREAD

**Discussion by kahboom** [KaotoIO/kaoto-ui#1643] [Announcements]
## Overview
This marks the first major release for Kaoto. 🍾  Kaoto is an open source visual integration and workflow editor. It supports multiple DSLs and advanced integration capabilities for both developers and non-developers.

Kaoto was built to be extended, and it gives you a layer of abstraction where you can store catalogs of steps/actions and custom configuration views for those steps/actions outside of Kaoto.

Kaoto is also available as a [VSCode extension](https://marketplace.visualstudio.com/items?itemName=redhat.vscode-kaoto).

![Kaoto Kafka configuration](https://user-images.githubusercontent.com/3844502/231195329-5f50b1f9-3c65-4f1f-a37b-43108e0f807b.png)

![Kaoto Camel component step catalog](https://user-images.githubusercontent.com/3844502/231195876-8aab6020-86f6-4ae6-96b2-09f58da53e24.png)

![Camel Content based router EIP configuration](https://user-images.githubusercontent.com/3844502/231195686-8079507f-4903-401d-b689-353255591b75.png)

![Kafka topic configuration help](https://user-images.githubusercontent.com/3844502/231195370-d007f192-b138-4487-ab91-41557fbdb78a.png)

![Kaoto dark mode](https://user-images.githubusercontent.com/3844502/231195386-ab46154e-ab95-4acb-b778-11a6da088be7.png)

### Features
- Drag and drop & point and click support of building integrations visually
- Choice of displaying the visualization in horizontally/vertically
- Searchable visual catalog of actions/steps (typically Camel components)
- One-click synchronization of canvas (visual integration) and its corresponding code
- Validation of YAML syntax
- Validation of Camel-specific syntax through Camel language server protocol
- One-click deployment of YAML-based integrations or custom resources to the cloud, with a specific namespace
- Upload/download YAML code for integrations
- Guided hints with tooltips throughout workflow
- Simple visualization of EIPs and multiple execution flows with branching, including built-in support for validation rules (i.e. minimum/maximum branches)
- Support for working with multiple DSLs (e.g. KameletBinding, Camel Components)
- Automatic generation of forms for action/step/component properties
- Support for extending each action/step/component with its own custom configuration view
- View, search, and manage deployments from a single place
- View deployment output through a dedicated log
- Dark mode

### Limitations
- Expression is limited to constant, simple and jq. Others will come later - https://github.com/KaotoIO/kaoto-backend/issues/516
- Some syntax of Camel YAML DSL is not yet supported in the source code editor. If there's unsupported syntax in the source code, it could be removed when you click sync your code
- Stop step connects back to the parent - https://github.com/KaotoIO/kaoto-ui/issues/1587

## What's Changed on the Frontend
Since the last release (`v1.0.0-rc1`), changes primarily includes stabilization of the code through priority bug fixes and tests.

### Fixes
* fix(steps-details): Required object type's parameters are lost by @lordrip in https://github.com/KaotoIO/kaoto-ui/pull/1605
* fix: yarn build:lib on CI related to OTEL deps by @helio-frota in https://github.com/KaotoIO/kaoto-ui/pull/1607
* fix: add stepNode__clickable class to Node by @unsortedhashsets in https://github.com/KaotoIO/kaoto-ui/pull/1615
* fix: replace "null" string with "No default value" by @unsortedhashsets in https://github.com/KaotoIO/kaoto-ui/pull/1614

### Everything Else
* chore: prepare next development iteration 0.7.7 by @apupier in https://github.com/KaotoIO/kaoto-ui/pull/1596
* chore(test): E2E finish refactoring by @unsortedhashsets in https://github.com/KaotoIO/kaoto-ui/pull/1593
* chore(test): add unit tests for all stores by @kahboom in https://github.com/KaotoIO/kaoto-ui/pull/1606
* chore: updating to version 1.0.0-rc1 by @igarashitm in https://github.com/KaotoIO/kaoto-ui/pull/1625
* chore(test): stabilized e2e tests for PR's by @tplevko in https://github.com/KaotoIO/kaoto-ui/pull/1634
* chore(test): add tests for stepsService & Kaoto UI API by @kahboom in https://github.com/KaotoIO/kaoto-ui/pull/1635
* chore: release 1.0.0 🎉 by @kahboom in https://github.com/KaotoIO/kaoto-ui/pull/1642

## Contributors
Thank you to everyone who made this release possible, whether by a code contribution, feedback, advocacy, or participating in an important discussion with us. ❤️

@aaj013 @almen-ng @amysueg @apupier @astefanutti @balellu @balejosg @balejos @brunoNetId @brunopedrosobuilders @caponetto @christophd @Croway @danielezonca @delawen @djelinek @ederign @evanshortiss @erayerdem @GuilhermeCamposo @helio-frota @hguerrero @igarashitm @ibek @javierlopezdeancos @jcordes73 @joshiraez @kahboom @krisv @lazarotti @lhein @lordrip @lburgazzoli @Mdenisco @mmelko @miram13 @MohammadiIram @pamenon @phantomjinx @Pompette @prapthik @rstroop @squakez @tadayosi @thiagoelg @tiagobento @timothydilbert @tplevko @unsortedhashsets @varkrish @zbendhiba 

And even the bots: @dependabot @deepsourcebot

Apologies in advance if I've missed anyone. 🙂 

See Kaoto backend release notes for v1.0.0 here: https://github.com/KaotoIO/kaoto-backend/releases/tag/v1.0.0

**Full Changelog**: https://github.com/KaotoIO/kaoto-ui/compare/v0.7.6...v1.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/KaotoIO/kaoto-ui/releases/tag/v1.0.0'>v1.0.0 🎉</a>.</em>

**Comment by heiko-braun** [comment]
Awesome! Congratulations everyone!!
